### PR TITLE
WiX: Implement auto-update.

### PR DESCRIPTION
**Pull request by mook-as**
This implements auto-update for Windows Installer:
- When examining releases, look for *.msi instead of *.exe
- Automatically determine if elevation is required
- Execute msiexec on application exit

In addition, we now build the Windows Installer package as part of `npm run build`.

Both of these only happen if a `RD_FEAT_WIX` environment variable is set (to `1`) during the build.

For testing, please override `upgradeServer`, `owner`, and `repo` in [`electron-builder.yml`](https://github.com/rancher-sandbox/rancher-desktop/blob/main/electron-builder.yml).

Due to some indenting issues in `main.wxs`, it may be better to use `diff -w` (etc.) when looking at changes.

### CODE REVIEW

File: pkg/rancher-desktop/main/update/MSIUpdater.ts [hunk 1]
```diff
@@ -0,0 +1,154 @@
...
+    super(options, app);
+  }
+
+  // This implements an abstract method in BaseUpdater.
+  protected doDownloadUpdate(downloadUpdateOptions: DownloadUpdateOptions): Promise<string[]> {
+    const { info, provider } = downloadUpdateOptions.updateInfoAndProvider;
+    const fileInfo = findFile(provider.resolveFiles(info), 'msi');
+
+    if (!fileInfo) {
+      throw newError(`Could not find update information for MSI installer`,
```
Comment (Nino-K) [inline pkg/rancher-desktop/main/update/MSIUpdater.ts:36]: Would it be helpful to include `info` in the error? is there anything useful in the `info`?

File: pkg/rancher-desktop/main/update/MSIUpdater.ts [hunk 2]
```diff
@@ -0,0 +1,154 @@
...
+
+    this._logger.debug?.(`Will invoke installer on restart with: msiexec ${ args.join(' ') }`);
+    mainEvents.on('quit', () => {
+      this._logger.debug?.(`Running msiexec ${ args.join(' ') }`);
+      const proc = spawn(msiexec, args, {
+        detached: true, stdio: 'ignore', windowsHide: true,
+      });
+
+      proc.on('error', (err: NodeJS.ErrnoException) => {
+        this._logger.info(`Cannot run installer: error code: ${ err.code }`);
```
Comment (Nino-K) [inline pkg/rancher-desktop/main/update/MSIUpdater.ts:117]: Is this meant to be an error or info?

**Review by Nino-K — COMMENTED** [review]
Mainly looks good but requires clarification on a few things.

File: pkg/rancher-desktop/main/update/MSIUpdater.ts [hunk 3]
```diff
@@ -0,0 +1,154 @@
...
+        this.dispatchError(err);
+      });
+      proc.on('exit', (code, signal) => {
+        this._logger.debug?.(`msiexec exited with ${ code }/${ signal }`);
+      });
+      proc.unref();
+    });
+
+    return true;
```
[the remark this answers is not in the document]
Comment (mook-as) [reply pkg/rancher-desktop/main/update/MSIUpdater.ts:126]: Yes, it always returns true; it's implementing an abstract method from `BaseUpdater` with that signature.

File: pkg/rancher-desktop/main/update/MSIUpdater.ts (hunk 1 above)
Comment (mook-as) [reply pkg/rancher-desktop/main/update/MSIUpdater.ts]: Yes, `info.version` would be useful. Added in b8397db37c3b8e845e0779fa7796e495c00df7fc.

File: pkg/rancher-desktop/main/update/MSIUpdater.ts (hunk 2 above)
Comment (mook-as) [reply pkg/rancher-desktop/main/update/MSIUpdater.ts]: Oops, `error` would be more correct; fixed in b8397db37c3b8e845e0779fa7796e495c00df7fc.

File: pkg/rancher-desktop/main/update/MSIUpdater.ts [hunk 4]
```diff
@@ -0,0 +1,154 @@
...
+      // Unfortunately, doInstall is synchronous, so we need to use the sync
+      // version of access functions.
+      fs.accessSync(checkPath, fs.constants.F_OK);
+      this._logger.debug?.('Elevation is required.');
+
+      return true;
+    } catch (ex) {
+      this._logger.debug?.(`Elevation is not required: ${ ex }`);
+
+      return true;
```
[the remark this answers is not in the document]
Comment (mook-as) [reply pkg/rancher-desktop/main/update/MSIUpdater.ts]: Good catch; fixed in b8397db37c3b8e845e0779fa7796e495c00df7fc.

**Review by Nino-K — APPROVED** [review]
LGTM, I think @mook-as might need to stash the last commit: b8397db37c3b8e845e0779fa7796e495c00df7fc

**Review by Nino-K — APPROVED** [review]

**Pull request merged**